### PR TITLE
Add a 'my pushes' view and add the possibility to filter pushes by state

### PR DESCRIPTION
**Pull request by sjaensch**
The 'my pushes' button filters the list of pushes by pushmaster, showing only the pushes you ran. Also, we added a drop down to filter the list of pushes by state. These two filters can be combined and will be preserved through pagination.

Screenshot of the UI:

### CODE REVIEW

**Comment by sjaensch** [comment]
I've updated the branch and fixed the pagination. It was slightly broken for some time, using the push ID for paginating while ordering the list by modified date. This meant that sometimes pagination links were shown even though they shouldn't be, since the first element in the list is not always the one with the highest ID, and the last element is not always the one with the lowest ID.

I have changed it to a traditional offset based pagination, using a SQL count() to determine the number of items we're paginating over. While this doesn't scale for very large datasets, it should be no issue at all for a list with a couple of thousand items.

**Comment by ymilki** [comment]
Can you split these two features into two different requests?

**Comment by sjaensch** [comment]
Done, posted the state filter changes as well as the pagination fix as #155. Will update this PR once the other one is merged since they touch the same code.

**Comment by sjaensch** [comment]
Merged master and resolved conflicts, also further improved the pushes API tests a bit. Here are some screenshots. We could remove that button if you feel it clutters the UI and just leave the functionality in, so that pushmasters could filter by changing the URL.

Filtered by user:
filtered by user

Filtered by user and state:
filtered by user and state

File: pushmanager/templates/pushes.html [hunk 1]
```diff
@@ -9,6 +9,16 @@
 <ul id="action-buttons">
 	<li><button id="expand-pushes">Expand All</button></li>
 	<li><button id="collapse-pushes">Collapse All</button></li>
+		&mdash;
+	<li>
+		Show states:
```
Comment (kkellyy) [inline pushmanager/templates/pushes.html]: This looks like it's a duplicate of the lines below. Bad merge resolution, maybe?

Comment (ymilki) [inline pushmanager/templates/pushes.html]: This Show states appears to be duplicated but the original is certainly not in master. This diff is weird.

Comment (sjaensch) [inline pushmanager/templates/pushes.html]: Sorry, my bad - missed that! It's indeed a bad merge resolution.

**Comment by kkellyy** [comment]
Cool. Looks good to me now. I'll merge this in soon.

**Pull request merged**